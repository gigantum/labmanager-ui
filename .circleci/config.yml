version: 2
jobs:
  build:
    docker:
      - image: gigantum/circleci-service-client:992bcbab-2018-07-09
    steps:
      - checkout
      - setup_remote_docker
      - run:
          name: Initialize submodules
          command: |
            git submodule update --init --recursive
      - run:
          name: Build frontend production code
          command: |
            npm install
            npm install relay
            npm run relay
            unset CI && npm run build
      - run:
          name: Initialize submodules
          command: |
            jest dashboard -u
