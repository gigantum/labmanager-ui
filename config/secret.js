const secrets = {
  accessToken: 'eyJ0eXAiOiJKV1QiLCJhbGciOiJSUzI1NiIsImtpZCI6Ik56a3dRekV5TVRORFJVTXpRakkzUmtRMlFrVkVRME0xT0RCRE5qRXhNalJHT1VOQ1JEQkJRZyJ9.eyJpc3MiOiJodHRwczovL2dpZ2FudHVtLmF1dGgwLmNvbS8iLCJzdWIiOiJhdXRoMHw1OWU4YzgyYmMzMGEzODA1M2FiNTM2YjMiLCJhdWQiOlsiYXBpLmdpZ2FudHVtLmlvIiwiaHR0cHM6Ly9naWdhbnR1bS5hdXRoMC5jb20vdXNlcmluZm8iXSwiaWF0IjoxNTI3MDg4Mzg1LCJleHAiOjE1MjcxNzQzODUsImF6cCI6IjBBam84T3Y2UXJpYWZvMHhWcFN5ZWpXeTFwd05XZGhrIiwic2NvcGUiOiJvcGVuaWQgcHJvZmlsZSBlbWFpbCJ9.QHxIdCtoHr1DgAmyYhxQ70J2p0j0HFBDAuCPPT_Vr6yVFsyJNMvYLivedvgM7HJ2WcSgc84ZEeaG95SNkTl21wA4GnpP2vLJc82DecFm-iDM5ip-6fvW9-Gr0HTRIldfAe-qtiLy70DMbBuQW7zqEMAs8FHIn8zqPpG0AD9toGBqXWlX-wCCIVobG9ACHrzWp5HpQmnxtN1tXEFIdfMv6xdFM5aKhcNLf8WEWRxueOdXvD697tB921bxWpxIFqy-pKiAsHK5hMc0JT9e1DLwqnsu6Qt2TzA69hI4mTyhfRZiemrRY82yUggCMyBjC1YjtZMjKem2g3CHmnFR-P5bGQ',
  'username': 'front-end-test'
}

export default secrets
