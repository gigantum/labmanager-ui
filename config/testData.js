import developerConfig

export const testData = {
  createLabbookMutation:{
    labbookName: 'ui-test-automated',
    'owner': 'gigantum_test'
  }
}
