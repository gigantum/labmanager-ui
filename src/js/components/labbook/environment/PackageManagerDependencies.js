//vendor
import React, { Component } from 'react'
import {createPaginationContainer, graphql} from 'react-relay'
//components
import AddEnvironmentPackage from 'Components/wizard/AddEnvironmentPackage'
import Loader from 'Components/shared/Loader'
let totalCount = 2
let owner
class PackageManagerDependencies extends Component {
  constructor(props){
    super(props);
    this.state = {
      'modal_visible': false
    };
    owner = this.props.owner

    this._openModal = this._openModal.bind(this)
    this._hideModal = this._hideModal.bind(this)
    this._setBaseImage = this._setBaseImage.bind(this)
    this._setComponent = this._setComponent.bind(this)
    this._loadMore = this._loadMore.bind(this)
  }
  /*
    handle state and addd listeners when component mounts
  */
  componentDidMount() {
    this._loadMore() //routes query only loads 2, call loadMore
  }
  /*
    @param
    triggers relay pagination function loadMore
    increments by 10
    logs callback
  */
  _loadMore() {

    let self = this;
    this.props.relay.loadMore(
     100, // Fetch the next 100 feed items
     (response, error) => {
       if(error){
         console.error(error)
       }

       if(self.props.environment.packageManagerDependencies &&
         self.props.environment.packageManagerDependencies.pageInfo.hasNextPage) {

         self._loadMore()
       }
     }
   );
  }
  /**
  *  @param {None}
  *  open modal sets state
  */
  _openModal = () =>{
      this.setState({'modal_visible': true})
      if(document.getElementById('modal__cover')){
        document.getElementById('modal__cover').classList.remove('hidden')
      }
  }
  /**
  *  @param {None}
  *  hides modal
  */
  _hideModal = () => {
      this.setState({'modal_visible': false})
      if(document.getElementById('modal__cover')){
        document.getElementById('modal__cover').classList.add('hidden')
      }

  }
  /**
  *  @param {None}
  *  set readyToBuild state to true
  */
  _setBaseImage = (baseImage) => {
    this.setState({"readyToBuild": true})
  }

  /**
  *  @param {Object}
  *  hides packagemanager modal
  */
  _setComponent = (comp) =>{
    // this.props.setContainerState("Building")
    // this.setState({"readyToBuild": true})
    this._hideModal();
  }

  render(){

    const {packageManagerDependencies} = this.props.environment;
    const {blockClass} = this.props;

    let editDisabled = ((this.props.containerStatus) && (this.props.containerStatus.state.imageStatus === "BUILD_IN_PROGRESS")) ? true : false;


    if (packageManagerDependencies) {
      return(
      <div className="Environment_package-manager-dependencies">
        <div className={!this.state.modal_visible ? 'Environment__modal hidden' : 'Environment__modal'}>
          <div
            id="packageManagerEditClose"
            className="Environment__modal-close"
            onClick={() => this._hideModal()}>
          </div>
          <AddEnvironmentPackage
            {...this.props}
            availablePackageManagers={(this.props.baseImage) ? this.props.baseImage.availablePackageManagers : null}
            setComponent={this._setComponent}
            nextComponent={"continue"}
            environmentView={true}
            toggleDisabledContinue={() => function(){}}
          />
        </div>

        <div className={blockClass + '__header-container'}>
          <h4 className="Environment__header">Package Dependencies</h4>
          <div className="Environment__edit-container">
            <button
              id="packageManagerEdit"
              className="Environment__edit-button"
              onClick={() => this._openModal()}
              disabled={editDisabled}
            >
            </button>
          </div>
        </div>

        <div className="Environment__info flex flex--row justify--left">
          <ul className="flex flex--row justify--left flex--wrap">
          {
            packageManagerDependencies.edges.map((edge, index) => {
              return(
                this._packageListItem(edge, index)
              )
            })
          }
        </ul>


      </div>
    </div>

      )
    }else{
      return(
          <Loader />
        )
    }
  }

  _packageListItem(edge, index){
    return(
      <li key={edge.packageName + edge.node.packageManager + index}>

          <div className="Environment__package-dependencies">

              <div className="Environment__card-text flex flex--row justify--space-around flex-1-0-auto">
                <p>{edge.node.packageManager}</p>
                <p>{edge.node.packageName}</p>
              </div>
          </div>

      </li>)
  }
}

export default createPaginationContainer(
  PackageManagerDependencies,
  {
    environment: graphql`fragment PackageManagerDependencies_environment on Environment {
    packageManagerDependencies(first: $first, after: $cursor) @connection(key: "PackageManagerDependencies_packageManagerDependencies" filters: []){
        edges{
          node{
            id
            packageManager
            packageName
            packageVersion
          }
          cursor
        }
        pageInfo{
          hasNextPage
          hasPreviousPage
          startCursor
          endCursor
        }
      }
    }`
  },
  {
    direction: 'forward',
    getConnectionFromProps(props) {

        return props.environment && props.environment.packageManagerDependencies;
    },
    getFragmentVariables(prevVars, first) {
      return {
       ...prevVars,
       first: first,
     };
   },
   getVariables(props, {count}, fragmentVariables) {

    totalCount += count
    let first = totalCount;
    let length = props.environment.packageManagerDependencies.edges.length
    let name = props.labbookName;

    let cursor = props.environment.packageManagerDependencies.edges[length-1].cursor

     return {
       first,
       cursor,
       name,
       owner
       // in most cases, for variables other than connection filters like
       // `first`, `after`, etc. you may want to use the previous values.
       //orderBy: fragmentVariables.orderBy,
     };
   },
   query: graphql`
    query PackageManagerDependenciesPaginationQuery($name: String!, $owner: String!, $first: Int!, $cursor: String){
     labbook(name: $name, owner: $owner){
       environment{
         ...PackageManagerDependencies_environment
       }
     }
   }`
}
)
