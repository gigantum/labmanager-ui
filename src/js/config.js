import tips from './tips'

const CONFIG = {
  'api': process.env.NODE_ENV,
  'navigation_items': [
    {id:'overview', name: 'Overview'},
    {id:'activity', name: 'Activity', 'fragment': '...Activity_labbook'},
    {id:'environment', name: 'Environment', 'fragment': '...Environment_labbook'},
    {id:'code', name: 'Code'},
    {id:'inputData', name: 'Input Data'},
    {id:'outputData', name: 'Output Data'}
  ],
  'modalNav': [
    {'id': 'createLabook', 'description': 'Title & Description'},
    {'id': 'selectBaseImage', 'description': 'Base Image'},
    {'id': 'selectDevelopmentEnvironment', 'description': 'Dev Environment'},
    {'id': 'addEnvironmentPackage', 'description': 'Add Dependencies'},
    {'id': 'successMessage', 'description': 'Success'}
  ],
  'months': [
    'Jan',
    'Feb',
    'Mar',
    'Apr',
    'May',
    'Jun',
    'Jul',
    'Aug',
    'Sep',
    'Oct',
    'Nov',
    'Dec'
  ],
  'fileBrowser':{
    'excludedFiles': [
      'DS_Store',
      'zip',
      'lbk',
      'pyc',
      'gitkeep'
    ]
  },

  containerStatus:{
    canEditEnvironment: (status) => {
      return (status === 'Stopped') || (status === 'Rebuild')
    }
  },
  userAPI: {
    getUsersQueryString: (userInput) => {
      const sanitizedUserInput = userInput.replace(/-/g, ' ')
      const apiURL = `https://m9eq4m3z0f.execute-api.us-east-1.amazonaws.com/prod?q=${sanitizedUserInput}*&q.options={fields: ['username^5','name']}&size=10`

      return encodeURI(apiURL)
    },
    getUserEmailQueryString: (email) => {
      const apiURL = `https://m9eq4m3z0f.execute-api.us-east-1.amazonaws.com/prod?q=${email}&q.options={fields: ['email']}&size=10`

      return encodeURI(apiURL)
    }
  },
  getToolTipText: (section) =>{
    return tips[section]
  },
  demoHostName: 'try.gigantum.com',
}

export default CONFIG
