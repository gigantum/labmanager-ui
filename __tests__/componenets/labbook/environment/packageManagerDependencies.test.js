import React from 'react';
import PackageManagerDependencies from 'Components/labbook/environment/PackageManagerDependencies';
import renderer from 'react-test-renderer';
import config from './../config'

let _setComponent = () => ({});
let _setBaseImage = () => ({});
let _buildCallback = () => ({});

test('Test PakageManager rendering', () => {

  //const isAuthenticated = function(){return true};
  const component = renderer.create(
    <PackageManagerDependencies
      environment={config.data.labbook.environment}
      blockClass={"Environment"}
      labbookName={config.data.labbook.name}
      environmentId={config.data.labbook.environment.id}
      setBaseImage={_setBaseImage}
      setComponent={_setComponent}
      buildCallback={_buildCallback}
      baseImage={config.data.labbook.environment.baseImage}
    />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();

});
