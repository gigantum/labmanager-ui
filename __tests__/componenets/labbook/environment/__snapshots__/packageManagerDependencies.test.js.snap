// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test PakageManager rendering 1`] = `
<div
  className="Environment_package-manager-dependencies"
>
  <div
    className="Environment__modal hidden"
  >
    <div
      className="Environment__modal-close"
      onClick={[Function]}
    >
      X
    </div>
    <div
      className="AddEnvironmentPackage"
    >
      <div
        className="AddEnvironmentPackage__inner-container"
      >
        <p>
          Install dependencies via Package Manager
        </p>
        <div
          className="AddEnvironmentPackage__row flex flex--row justify--space-between"
        >
          <div
            className="AddEnvironmentPackage__select--container"
          >
            <select
              className="AddEnvironmentPackage__select"
              disabled={false}
              onChange={[Function]}
            >
              <option
                className="AddEnvironmentPackage__select"
                value="pip3"
              >
                pip3
              </option>
              <option
                className="AddEnvironmentPackage__select"
                value="apt-get"
              >
                apt-get
              </option>
            </select>
          </div>
          <input
            autoFocus={true}
            className="AddEnvironmentPackage__text-input"
            disabled={false}
            onKeyUp={[Function]}
            placeholder="Dependency Name"
            type="text"
          />
          <button
            className="AddEnvironmentPackage__button"
            disabled={true}
            onClick={[Function]}
          >
            Add
          </button>
        </div>
      </div>
      <div
        className="AddEnvironmentPackage__progress-buttons flex flex--row justify--space-between"
      >
        <button
          onClick={[Function]}
        >
          Save
        </button>
      </div>
    </div>
  </div>
  <h4
    className="Environment__header"
  >
    Package Dependencies
  </h4>
  <div
    className="Environment__info flex flex--row justify--left"
  >
    <ul
      className="flex flex--row justify--left flex--wrap"
    >
      <li>
        <div
          className="Environment__package-dependencies"
        >
          <div
            className="Environment__card-text flex flex--row justify--space-around flex-1-0-auto"
          >
            <p>
              pip3
            </p>
            <p>
              numpy
            </p>
          </div>
        </div>
      </li>
      <li>
        <div
          className="Environment__package-dependencies"
        >
          <div
            className="Environment__card-text flex flex--row justify--space-around flex-1-0-auto"
          >
            <p>
              pip3
            </p>
            <p>
              nuumpy
            </p>
          </div>
        </div>
      </li>
      <li>
        <div
          className="Environment__package-dependencies"
        >
          <div
            className="Environment__card-text flex flex--row justify--space-around flex-1-0-auto"
          >
            <p>
              pip3
            </p>
            <p>
              plotly
            </p>
          </div>
        </div>
      </li>
    </ul>
    <div
      className="Environment__edit-container"
    >
      <button
        className="Environment__edit-button"
        onClick={[Function]}
      >
        Edit
      </button>
    </div>
  </div>
</div>
`;
